packages/zone.js/test/closure/zone.closure.ts

```diff
@@ -72,16 +72,57 @@ const testClosureFunction = () => {
       logs.push('getZoneWith' + keyZone!.name);
       logs.push('get' + keyZone!.get('key'));
       logs.push('root' + Zone.root.name);
-      Object.keys((Zone as any).prototype).forEach(key => {
-        logs.push(key);
+      const zonePrototypeKeys = [
+        'get',
+        'getZoneWith',
+        'fork',
+        'wrap',
+        'run',
+        'runGuarded',
+        'runTask',
+        'scheduleTask',
+        'scheduleMicroTask',
+        'scheduleMacroTask',
+        'scheduleEventTask',
+        'cancelTask',
+      ];
+      zonePrototypeKeys.forEach(key => {
+        const desc = Object.getOwnPropertyDescriptor((Zone as any).prototype, key);
+        if (desc) {
+          logs.push(key);
+        }
       });
-      Object.keys(testZoneSpec).forEach(key => {
-        logs.push(key);
+
+      const zoneSpecKeys = [
+        'name',
+        'properties',
+        'onFork',
+        'onIntercept',
+        'onInvoke',
+        'onHandleError',
+        'onScheduleTask',
+        'onInvokeTask',
+        'onCancelTask',
+        'onHasTask',
+      ];
+      zoneSpecKeys.forEach(key => {
+        const desc = Object.getOwnPropertyDescriptor(testZoneSpec, key);
+        if (desc) {
+          logs.push(key);
+        }
       });
 
+      const zoneTaskKeys = [
+        'onHasTask', 'runCount', 'type', 'source', 'data', 'scheduleFn', 'cancelFn', 'callback',
+        'invoke'
+      ];
+
       const task = Zone.current.scheduleMicroTask('testTask', () => {}, undefined, () => {});
-      Object.keys(task).forEach(key => {
-        logs.push(key);
+      zoneTaskKeys.forEach(key => {
+        const desc = Object.getOwnPropertyDescriptor(task, key);
+        if (desc) {
```

Comment:
Would `hasOwnProperty()` be simpler?

```suggestion
        if (task.hasOwnProperty(key)) {
```

Reply:
Yes, you are right.